parcels/_index_search.py

```diff
@@ -37,8 +37,9 @@ def _search_time_index(field: Field, time: datetime, allow_time_extrapolation=Tr
     Note that we normalize to either the first or the last index
     if the sampled value is outside the time value range.
     """
-    if not allow_time_extrapolation and (time < field.data.time[0] or time > field.data.time[-1]):
+    if (len(field.data.time)) > 1 and (time < field.data.time[0] or time > field.data.time[-1]):
         _raise_time_extrapolation_error(time, field=None)
+
```

Comment:
I think it would be easier if we do something like `if Field.time_interval and time in Field.time_interval:`. I just put up #2000 which does the groundwork for this

Reply:
Made the change in https://github.com/OceanParcels/Parcels/pull/1999/commits/14ef097f8c45a7d50fbdba25b3b7d61806e0288b